bittensor/utils/async_substrate.py

```diff
@@ -31,41 +60,340 @@ class Preprocessed:
-    async def wrapper(self, *args, **kwargs):
+    def reload_type_registry(
+        self, use_remote_preset: bool = True, auto_discover: bool = True
+    ):
+        """
+        Reload type registry and preset used to instantiate the SubstrateInterface object. Useful to periodically apply
+        changes in type definitions when a runtime upgrade occurred
+
+        Parameters
+        ----------
+        use_remote_preset: When True preset is downloaded from Github master, otherwise use files from local installed scalecodec package
+        auto_discover
+
+        Returns
+        -------
+
+        """
+        self.runtime_config.clear_type_registry()
+
+        self.runtime_config.implements_scale_info = self.implements_scaleinfo
+
+        # Load metadata types in runtime configuration
+        self.runtime_config.update_type_registry(load_type_registry_preset(name="core"))
+        self.apply_type_registry_presets(
+            use_remote_preset=use_remote_preset, auto_discover=auto_discover
+        )
+
+    def apply_type_registry_presets(
+        self,
+        use_remote_preset: bool = True,
+        auto_discover: bool = True,
+    ):
+        if self.type_registry_preset is not None:
+            # Load type registry according to preset
+            type_registry_preset_dict = load_type_registry_preset(
+                name=self.type_registry_preset, use_remote_preset=use_remote_preset
+            )
+
+            if not type_registry_preset_dict:
+                raise ValueError(
+                    f"Type registry preset '{self.type_registry_preset}' not found"
+                )
+
+        elif auto_discover:
+            # Try to auto discover type registry preset by chain name
+            type_registry_name = self.chain.lower().replace(" ", "-")
+            try:
+                type_registry_preset_dict = load_type_registry_preset(
+                    type_registry_name
+                )
+                self.type_registry_preset = type_registry_name
+            except ValueError:
+                type_registry_preset_dict = None
+
+        else:
+            type_registry_preset_dict = None
+
+        if type_registry_preset_dict:
+            # Load type registries in runtime configuration
+            if self.implements_scaleinfo is False:
+                # Only runtime with no embedded types in metadata need the default set of explicit defined types
+                self.runtime_config.update_type_registry(
+                    load_type_registry_preset(
+                        "legacy", use_remote_preset=use_remote_preset
+                    )
+                )
+
+            if self.type_registry_preset != "legacy":
+                self.runtime_config.update_type_registry(type_registry_preset_dict)
+
+        if self.type_registry:
+            # Load type registries in runtime configuration
+            self.runtime_config.update_type_registry(self.type_registry)
+
+
+class RequestManager:
```

Comment:
I remember we discussed this, but I would like to return to the question of using async class methods. 
We want to have async across whole bittensor. 
A good practice in async development is to take the approach: `if you can use async call instead of sync one, do it`.

I don’t agree that this complicates the code, but rather confuses it. If I were not familiar with the code, then I would have a question: “Why is sync code used exact here if the entire SDK is async?”

Reply:
It's the same reason we don't do asyncio dictionaries. Asyncio only matters when there is I/O. There is no I/O in RequestManager.

Reply:
Considering that we do not store complex data there, most likely I will take your side in this.